File: android/services-v72/src/main/java/com/liferay/mobile/android/v72/powwowmeeting/PowwowmeetingService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v72.powwowmeeting;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class PowwowmeetingService extends BaseService {

	public PowwowmeetingService(Session session) {
		super(session);
	}

	public JSONObject addPowwowMeeting(long groupId, String portletId, long powwowServerId, String name, String description, String providerType, JSONObject providerTypeMetadataMap, String languageId, long calendarBookingId, int status, JSONArray powwowParticipants, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("calendarBookingId", calendarBookingId);
			_params.put("description", checkNull(description));
			_params.put("groupId", groupId);
			_params.put("languageId", checkNull(languageId));
			_params.put("name", checkNull(name));
			_params.put("portletId", checkNull(portletId));
			_params.put("powwowParticipants", checkNull(powwowParticipants));
			_params.put("powwowServerId", powwowServerId);
			_params.put("providerType", checkNull(providerType));
			_params.put("providerTypeMetadataMap", checkNull(providerTypeMetadataMap));
			_params.put("status", status);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/powwow-portlet.powwowmeeting/add-powwow-meeting", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject deletePowwowMeeting(long powwowMeetingId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("powwowMeetingId", powwowMeetingId);

			_command.put("/powwow-portlet.powwowmeeting/delete-powwow-meeting", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject getPowwowMeeting(long powwowMeetingId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("powwowMeetingId", powwowMeetingId);

			_command.put("/powwow-portlet.powwowmeeting/get-powwow-meeting", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getPowwowMeetings(long groupId, int start, int end, JSONObjectWrapper obc) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("end", end);
			_params.put("groupId", groupId);
			_params.put("start", start);
			mangleWrapper(_params, "obc", "com.liferay.portal.kernel.util.OrderByComparator", obc);

			_command.put("/powwow-portlet.powwowmeeting/get-powwow-meetings", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getPowwowMeetingsCount(long groupId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("groupId", groupId);

			_command.put("/powwow-portlet.powwowmeeting/get-powwow-meetings-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject updatePowwowMeeting(long powwowMeetingId, long powwowServerId, String name, String description, String providerType, JSONObject providerTypeMetadataMap, String languageId, long calendarBookingId, int status, JSONArray powwowParticipants, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("calendarBookingId", calendarBookingId);
			_params.put("description", checkNull(description));
			_params.put("languageId", checkNull(languageId));
			_params.put("name", checkNull(name));
			_params.put("powwowMeetingId", powwowMeetingId);
			_params.put("powwowParticipants", checkNull(powwowParticipants));
			_params.put("powwowServerId", powwowServerId);
			_params.put("providerType", checkNull(providerType));
			_params.put("providerTypeMetadataMap", checkNull(providerTypeMetadataMap));
			_params.put("status", status);
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/powwow-portlet.powwowmeeting/update-powwow-meeting", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}

File: windows/Liferay.SDK/Service/V72/Powwowmeeting/PowwowmeetingService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V72.Powwowmeeting
{
	public class PowwowmeetingService : ServiceBase
	{
		public PowwowmeetingService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> DeletePowwowMeetingAsync(long powwowMeetingId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("powwowMeetingId", powwowMeetingId);

			var _command = new JsonObject()
			{
				{ "/powwow-portlet.powwowmeeting/delete-powwow-meeting", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> UpdatePowwowMeetingAsync(long powwowMeetingId, long powwowServerId, string name, string description, string providerType, IDictionary<string, object> providerTypeMetadataMap, string languageId, long calendarBookingId, int status, IEnumerable<object> powwowParticipants, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("powwowMeetingId", powwowMeetingId);
			_parameters.Add("powwowServerId", powwowServerId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("providerType", providerType);
			_parameters.Add("providerTypeMetadataMap", providerTypeMetadataMap);
			_parameters.Add("languageId", languageId);
			_parameters.Add("calendarBookingId", calendarBookingId);
			_parameters.Add("status", status);
			_parameters.Add("powwowParticipants", powwowParticipants);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/powwow-portlet.powwowmeeting/update-powwow-meeting", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetPowwowMeetingAsync(long powwowMeetingId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("powwowMeetingId", powwowMeetingId);

			var _command = new JsonObject()
			{
				{ "/powwow-portlet.powwowmeeting/get-powwow-meeting", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddPowwowMeetingAsync(long groupId, string portletId, long powwowServerId, string name, string description, string providerType, IDictionary<string, object> providerTypeMetadataMap, string languageId, long calendarBookingId, int status, IEnumerable<object> powwowParticipants, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("portletId", portletId);
			_parameters.Add("powwowServerId", powwowServerId);
			_parameters.Add("name", name);
			_parameters.Add("description", description);
			_parameters.Add("providerType", providerType);
			_parameters.Add("providerTypeMetadataMap", providerTypeMetadataMap);
			_parameters.Add("languageId", languageId);
			_parameters.Add("calendarBookingId", calendarBookingId);
			_parameters.Add("status", status);
			_parameters.Add("powwowParticipants", powwowParticipants);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/powwow-portlet.powwowmeeting/add-powwow-meeting", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<long> GetPowwowMeetingsCountAsync(long groupId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);

			var _command = new JsonObject()
			{
				{ "/powwow-portlet.powwowmeeting/get-powwow-meetings-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetPowwowMeetingsAsync(long groupId, int start, int end, JsonObjectWrapper obc)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "obc", "$languageUtil.getJSONWrapperClassName($parameter.type)", obc);

			var _command = new JsonObject()
			{
				{ "/powwow-portlet.powwowmeeting/get-powwow-meetings", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}
	}
}